test/test_matmul_cuda.py

```diff
@@ -268,8 +268,9 @@ def grouped_mm_helper(self, alist, blist, gOlist, agradlist, bgradlist, outlist)
             out_ref = torch.mm(a, b.t())
             out_ref.backward(gO)
             self.assertEqual(out, out_ref)
-            self.assertEqual(agrad, a.grad)
-            self.assertEqual(bgrad, b.grad)
+            if agrad is not None:
+                self.assertEqual(agrad, a.grad)
+                self.assertEqual(bgrad, b.grad)
 
     @unittest.skipIf(TEST_WITH_ROCM, "ROCm doesn't support CUTLASS")
     @xfailIfSM100OrLater
```

Comment:
I'm surprised this did not fire as a type error:

![image](https://github.com/user-attachments/assets/a8dd4aab-e748-48b4-ba0d-835a1c40f589)


I think the expressions in https://github.com/pytorch/pytorch/blob/d808a3e203da8464ea0da623716a6098f7655ba9/torch/_inductor/kernel/mm_scaled_grouped.py#L370 should be using the statically_known variants from sizevars